client/react/src/components/grades/grades-view.tsx

```diff
@@ -0,0 +1,139 @@
+"use client";
+
+import { useState } from "react";
+import { ChevronDown } from "lucide-react";
+import { degrees } from "./mock-data";
+import { Degree, Semester, Module } from "./types";
+import clsx from "clsx";
+import { Card } from "@/components/ui/card";
+import { Separator } from "@/components/ui/separator";
+
+// TODO: reafactor this component and split into smaller components
+
+const GradesView = () => {
+  return (
+    <Card className="font-mono text-sm min-w-full max-w-5xl mx-auto p-4 rounded-lg">
+      <div className="grid grid-cols-2 mb-2 px-4">
+        <span>Name</span>
+        <span className="text-right">ECTS & Grade</span>
+      </div>
+      <div className="space-y-2">
+        {degrees.map((degree, index) => {
+          // if is last return without separator
```

Comment:
What do you think about the idea to outsource the logic in a separate file to enhance the readability?

Reply:
the degrees.map logic i think we can leave but the other components (DegreeSection, SemesterSection, ...) should be outsourced in an upcoming refactor

Reply:
I agree with @mvoemel.